SGoopas/Assets/Scripts/MasterStateMachine.cs

```diff
@@ -83,32 +90,39 @@ private IEnumerator UpdateLoadingWithProgress(AsyncOperation operation)
         float progress;
         while (!operation.isDone)
         {
-            progress = Mathf.Clamp01(operation.progress / .9f);
-            MasterMonoBehaviour.Instance.slider.value = progress;
-            MasterMonoBehaviour.Instance.progressTxt.text = progress * 100f + "%";
+			progress = Mathf.Clamp01(operation.progress / .9f);
+			MasterMonoBehaviour.Instance.slider.value = progress;
+			MasterMonoBehaviour.Instance.progressTxt.text = progress * 100f + "%";
+
             yield return null;
         }
+
+		MasterMonoBehaviour.Instance.FadeScreen(0);
     }
 
     public IEnumerator LoadLevelAsynchronously(GameMainState loading) {
-        MasterMonoBehaviour.Instance.loadScreen.SetActive(true);
```

Comment:
you can actually delete loadScreen.SetActive in these places now that you don't use it. Just make sure you keep SetAsActiveScene() and setState...